core/src/main/java/org/stellar/anchor/auth/JwtService.java

```diff
@@ -81,29 +85,31 @@ public String encode(Sep10Jwt token) {
     return builder.signWith(SignatureAlgorithm.HS256, sep10JwtSecret).compact();
   }
 
+  public String encode(Sep6MoreInfoUrlJwt token) throws InvalidConfigException {
+    if (sep6MoreInfoUrlJwtSecret == null) {
+      throw new InvalidConfigException(
+          "Please provide the secret before encoding JWT for Sep6 more_info_url");
+    }
+    return encodeForSep6Sep24Url(token, sep6MoreInfoUrlJwtSecret);
+  }
+
   public String encode(Sep24InteractiveUrlJwt token) throws InvalidConfigException {
     if (sep24InteractiveUrlJwtSecret == null) {
       throw new InvalidConfigException(
           "Please provide the secret before encoding JWT for Sep24 interactive url");
     }
-    Instant timeExp = Instant.ofEpochSecond(token.getExp());
-    JwtBuilder builder =
-        Jwts.builder()
-            .setId(token.getJti())
-            .setExpiration(from(timeExp))
-            .setSubject(token.getSub());
-    for (Map.Entry<String, Object> claim : token.claims.entrySet()) {
-      builder.claim(claim.getKey(), claim.getValue());
-    }
-
-    return builder.signWith(SignatureAlgorithm.HS256, sep24InteractiveUrlJwtSecret).compact();
+    return encodeForSep6Sep24Url(token, sep24InteractiveUrlJwtSecret);
   }
 
   public String encode(Sep24MoreInfoUrlJwt token) throws InvalidConfigException {
     if (sep24MoreInfoUrlJwtSecret == null) {
       throw new InvalidConfigException(
           "Please provide the secret before encoding JWT for more_info_url");
     }
+    return encodeForSep6Sep24Url(token, sep24MoreInfoUrlJwtSecret);
+  }
+
+  private String encodeForSep6Sep24Url(AbstractJwt token, String secret) {
```

Comment:
nit: What do you think of calling this `encodeForMoreInfoUrl`?

Reply:
nit: I would prefer `encodeForMoreInfoUrl` though because the function has nothing specific with sep6 or sep24. It just encode and sign the URL

Reply:
Ok  I will change it to `encodeForUr`l because it is also used to encode **interactive** url, so I won't put more_info in the naming